### PR TITLE
Cleanup and move architecture utils

### PR DESCRIPTION
**Pull request by sarranz**
General cleanup (prefer `Scheme Equality`) and move some more definitions to `arch_utils` for when I come back to OTBN. Should I use `t_eq_axiom` in other files? (or remove it?)

### CODE REVIEW

**Review by bgregoir — COMMENTED** [review]
I hope this will not do too many conflicts with #423...
Also by t_eq_axiom is a tactic and not a lemma ?

**Comment by sarranz** [comment]
> I hope this will not do too many conflicts with #423... Also by t_eq_axiom is a tactic and not a lemma ?
I can wait until we merge #423 if that helps, these changes are not hard to reproduce.
And that's right, it should be a lemma. I'll make this changes and use it everywhere.

**Review by vbgl — COMMENTED** [review]
This looks right.

**Review by vbgl — APPROVED** [review]

**Pull request merged**
